skimage/segmentation/_quickshift.pyx

```diff
@@ -2,20 +2,17 @@
 #cython: boundscheck=False
 #cython: nonecheck=False
 #cython: wraparound=False
-import numpy as np
-from scipy import ndimage as ndi
-from itertools import product
 
 cimport numpy as cnp
 from libc.math cimport exp, sqrt
 from libc.float cimport DBL_MAX
 
-from ..util import img_as_float
-from ..color import rgb2lab
 
-
-def quickshift(image, ratio=1.0, kernel_size=5, max_dist=10,
-               return_tree=False, sigma=0, convert2lab=True, random_seed=None):
+def _quickshift_cython(cnp.ndarray[float, ndim=3, mode="c"] image,
```

Comment:
When I declare `image` as memoryview here, I'm getting an error for the second line you commented.
Unfortunately, I have not enough knowledge about Cython yet to handle this case.